src/hotspot/share/runtime/thread.inline.hpp

```diff
@@ -204,13 +190,18 @@ inline bool JavaThread::is_terminated() const {
   return check_is_terminated(l_terminated);
 }
 
-inline void JavaThread::set_terminated(TerminatedTypes t) {
+inline void JavaThread::set_exiting() {
+  // use release-store so the setting of _terminated is seen more quickly
```

Comment:
Pre-existing: A release doesn't push changes to main memory more quickly it only establishes ordering with previous loads and stores. Why do we need release semantics here - what state are with ordering with? What load-acquire are we pairing with?

Reply:
I beilive this is legacy from pre-ThreadsList era.
Now we _should_ not have any such situations where such race matters.
A thread on a ThreadsList can become terminated but it is still perfectly reachable.
If we filter out terminated threads from a ThreadsList, any of those can become terminated just after any ways.

So Atomic::store/load is fine here IMHO.